### PR TITLE
Fix KeyError thrown during subscription cleanup

### PR DESCRIPTION
**Pull request by rjwills28**
This PR fixes the issue described in #2793 where a KeyError is thrown during cleanup of subscriptions using the graphql-ws-protocol. 
First check that the subscriptions dict contains the key before accessing. This has already been done a few lines down so simply move it to there.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #2793 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release fixes a bug causing a KeyError exception to be thrown during subscription cleanup.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to rjwills28 for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py [hunk 1]
```diff
@@ -276,10 +276,10 @@ async def operation_task(
             # cleanup in case of something really unexpected
             # wait for generator to be closed to ensure that any existing
             # 'finally' statement is called
-            result_source = self.subscriptions[operation.id]
-            with suppress(RuntimeError):
-                await result_source.aclose()
             if operation.id in self.subscriptions:
+                result_source = self.subscriptions[operation.id]
+                with suppress(RuntimeError):
+                    await result_source.aclose()
```
Comment (patrick91) [inline strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py:282]: would it be possible to get a test for this? or is too complicated to replicate?

**Comment by DavidLemayian** [comment]
Hi team. Any luck with this?

**Comment by patrick91** [comment]
@DavidLemayian sorry for the ping, I'll check this today!

**Comment by botberry** [comment]
# Pre-release
:wave:

Releasing commit [ab3d9d3ae5fac71ec491f53759aec143e828082c] to PyPi as pre-release! :package:

**Review by patrick91 — APPROVED** [review]
Just tested this in a project where I was able to replicate the issue, it worked fine!

Thanks 😊

**Comment by DavidLemayian** [comment]
Thank you @patrick91 ! 🥇 ❤️  🚀

**Pull request merged**